src/extensions/surveys/components/ConfirmationMessage.tsx

```diff
@@ -26,13 +28,22 @@ export function ConfirmationMessage({
                     <h3 className="thank-you-message-header" style={{ color: textColor }}>
                         {confirmationHeader}
                     </h3>
-                    {confirmationDescription && (
-                        <div
-                            style={{ color: textColor }}
-                            className="thank-you-message-body"
-                            dangerouslySetInnerHTML={{ __html: confirmationDescription }}
-                        />
-                    )}
+                    {confirmationDescription ? (
+                        confirmationDescriptionContentType === 'text' ? (
+                            <div style={{ color: textColor }} className="thank-you-message-body">
```

Comment:
I'd love this to be a helper so we don't have so much duplication here, this is very prone to errors. Something like:

`renderChildrenAsTextOrHtml({ component, children, renderAsHtml }) => // conditionally either add `dangerouslySetInnerHTML` on the component, or just add children as a child node`

Think you can use cloneelement for this https://preactjs.com/guide/v10/api-reference/#cloneelement - unless you find a better way ;)

Reply:
wrote a helper method for this that I used in both instances, seemed to work great!

Reply:
Amazing, we can reuse this now for when we extend this to other fields 🚀